devito/passes/iet/engine.py

```diff
@@ -398,18 +398,21 @@ def abstract_efunc(efunc):
     return efunc
 
 
-def abstract_objects(objects, sregistry=None):
+def abstract_objects(objects0, sregistry=None):
     """
     Proxy for `abstract_object`.
     """
+    # Expose hidden objects for complete reconstruction
+    objects = []
```

Comment:
If order doesn't matter, then deepcopying `objects0` or using `objects = list(object0)` instead of initialising an empty list would remove the `objects.append(i)` line